# Project 1: Finance Trends Analysis

**Author:** Haixin Liu

**Date:** November 2025

## Introduction 

Have you ever wondered how people make investment decisions? As someone 
interested in finance, I've always been curious about what drives investors 
to choose certain investment vehicles over others. 

When I discovered this Finance Trends dataset containing over 12,000 survey 
responses about investment preferences, I knew this was the perfect 
opportunity to explore these patterns. This dataset captures how real 
investors rate different investment options on a scale of 1-7, from mutual 
funds to gold investments.

**Dataset source:** [Kaggle – Finance Trends (2020–2025)](https://www.kaggle.com/datasets/ayeshasiddiqa123/finance-trends-2020-2025?resource=download)

**Dataset file:** [Finance_Trends.csv](Finance_Trends.csv)

**Size:** 12,000+ responses with 24 variables

### What I Want to Explore

In this analysis, I'll focus on understanding:
- How do investors rate different investment options?
- What is the typical (average) preference level?
- What is the most common rating given?
- Is there a pattern in how people evaluate investments?

### Why This Matters

Understanding investment preferences can help:
- Financial advisors better serve their clients
- Investment companies design better products  
- Individual investors understand market sentiment

Let's dive into the data and see what stories these 12,000 investors 
have to tell us!

In [6]:
import pandas as pd
import csv

In [ ]:
df = pd.read_csv("Finance_Trends.csv")
print(df.shape)
df.head()

(12000, 24)


gender  age Investment_Avenues  Mutual_Funds  Equity_Market  Debentures  \
0    Male   29                Yes             2              4           7   
1    Male   28                Yes             2              3           6   
2  Female   28                Yes             2              3           7   
3    Male   19                Yes             2              3           7   
4    Male   32                Yes             2              4           7   

   Government_Bonds  Fixed_Deposits  PPF  Gold  ...           Duration  \
0                 5               3    1     6  ...   Less than 1 year   
1                 5               1    4     7  ...   Less than 1 year   
2                 5               4    1     6  ...          3-5 years   
3                 4               6    1     5  ...  More than 5 years   
4                 5               3    1     6  ...  More than 5 years   

  Invest_Monitor   Expect                 Avenue  \
0         Weekly  20%-30%  Public Provident Fund   
1         Weekly  30%-40%  Public Provident Fund   
2          Daily  10%-20%            Mutual Fund   
3        Monthly  10%-20%                 Equity   
4         Weekly  10%-20%  Public Provident Fund   

  What are your savings objectives?         Reason_Equity  \
0                       Health Care              Dividend   
1                   Retirement Plan  Capital Appreciation   
2                       Health Care              Dividend   
3                   Retirement Plan  Capital Appreciation   
4                   Retirement Plan  Capital Appreciation   

          Reason_Mutual     Reason_Bonds      Reason_FD  \
0  Fund Diversification  Assured Returns  Fixed Returns   
1        Better Returns  Assured Returns      Risk Free   
2        Better Returns  Assured Returns      Risk Free   
3        Better Returns  Assured Returns      Risk Free   
4        Better Returns  Assured Returns  Fixed Returns   

                     Source  
0  Newspapers and Magazines  
1                Television  
2     Financial Consultants  
3  Newspapers and Magazines  
4     Financial Consultants  

[5 rows x 24 columns]

In [8]:
# Check column names and data types
print("Columns in the dataset:")
print(df.columns.tolist())
print("\nData types:")
print(df.dtypes)

Columns in the dataset:
['gender', 'age', 'Investment_Avenues', 'Mutual_Funds', 'Equity_Market', 'Debentures', 'Government_Bonds', 'Fixed_Deposits', 'PPF', 'Gold', 'Stock_Marktet', 'Factor', 'Objective', 'Purpose', 'Duration', 'Invest_Monitor', 'Expect', 'Avenue', 'What are your savings objectives?', 'Reason_Equity', 'Reason_Mutual', 'Reason_Bonds', 'Reason_FD', 'Source']

Data types:
gender                               object
age                                   int64
Investment_Avenues                   object
Mutual_Funds                          int64
Equity_Market                         int64
Debentures                            int64
Government_Bonds                      int64
Fixed_Deposits                        int64
PPF                                   int64
Gold                                  int64
Stock_Marktet                        object
Factor                               object
Objective                            object
Purpose                              obj

In [ ]:
# Select a numeric column for analysis
# Choose one of the investment rating columns (scale 1-7)
selected_column = df["Mutual_Funds"]  # or choose another column
print(f"Selected column for analysis: Mutual_Funds")
print(f"Data range: {selected_column.min()} to {selected_column.max()}")
print(f"Number of responses: {len(selected_column)}")

Selected column for analysis: Mutual_Funds
Data range: 1 to 7
Number of responses: 12000


## Part 2: Calculate Statistics with Pandas

Now let's calculate the three measures of central tendency:
- **Mean**: The average rating
- **Median**: The middle value when sorted
- **Mode**: The most frequent rating

In [10]:
# Calculate statistics using pandas
mean_pandas = selected_column.mean()
median_pandas = selected_column.median()
mode_result = selected_column.mode()

# Handle mode (might have multiple modes)
if len(mode_result) > 0:
    mode_pandas = mode_result[0]  # take first mode
else:
    mode_pandas = None

# Display results
print("Statistics from Pandas:")
print(f"Mean:   {mean_pandas:.4f}")
print(f"Median: {median_pandas:.4f}")
print(f"Mode:   {mode_pandas}")

# Let's see what these numbers tell us
print(f"\nInterpretation:")
print(f"The average rating of {mean_pandas:.2f} suggests...")
if mean_pandas > 4:
    print("  - Investors have a positive view of Mutual Funds")
else:
    print("  - Investors are neutral/cautious about Mutual Funds")

Statistics from Pandas:
Mean:   2.5346
Median: 2.0000
Mode:   2

Interpretation:
The average rating of 2.53 suggests...
  - Investors are neutral/cautious about Mutual Funds


In [11]:
# Check the distribution of ratings
rating_counts = selected_column.value_counts().sort_index()
print("Distribution of ratings:")
for rating, count in rating_counts.items():
    percentage = (count / len(selected_column)) * 100
    print(f"Rating {rating}: {count} responses ({percentage:.1f}%)")

Distribution of ratings:
Rating 1: 1202 responses (10.0%)
Rating 2: 6343 responses (52.9%)
Rating 3: 2743 responses (22.9%)
Rating 4: 852 responses (7.1%)
Rating 5: 565 responses (4.7%)
Rating 7: 295 responses (2.5%)


## Part 3: Manual Calculations 

Now let's calculate the same statistics WITHOUT using pandas or any 
statistics library. This helps us understand what's happening behind 
the scenes.

In [ ]:
# Read data manually
import csv

data = []
with open("Finance_Trends.csv", "r") as file:
    csv_reader = csv.DictReader(file)
    for row in csv_reader:
        data.append(int(row["Mutual_Funds"]))

print(f"Read {len(data)} values")

Read 12000 values


In [13]:
# Mean = sum / count
mean_manual = sum(data) / len(data)
print(f"Mean: {mean_manual:.4f}")

Mean: 2.5346


In [14]:
# Median = middle value
sorted_data = sorted(data)
n = len(sorted_data)

if n % 2 == 1:
    median_manual = sorted_data[n // 2]
else:
    median_manual = (sorted_data[n // 2 - 1] + sorted_data[n // 2]) / 2

print(f"Median: {median_manual:.4f}")

Median: 2.0000


In [15]:
# Mode = most frequent value (MUST USE DICTIONARY!)
counts = {}
for value in data:
    if value in counts:
        counts[value] += 1
    else:
        counts[value] = 1

# Find most common
mode_manual = max(counts, key=counts.get)
print(f"Mode: {mode_manual}")

Mode: 2


## Part 4: Data Visualization

Create visualizations using only the standard library - no matplotlib!
We'll use ASCII characters to create visual representations.

In [ ]:
# Create visual bar chart using symbols
print("\nRating Distribution Bar Chart:")
print("=" * 60)

for rating in range(1, 8):
    count = data.count(rating)
    percentage = (count / len(data)) * 100

    # Create visual bar with symbols
    bar_length = int(percentage / 2)  # Scale to fit
    bar = "█" * bar_length

    print(f"Rating {rating}: {bar} {count} ({percentage:.1f}%)")

print("=" * 60)


Rating Distribution Bar Chart:
Rating 1: █████ 1202 (10.0%)
Rating 2: ██████████████████████████ 6343 (52.9%)
Rating 3: ███████████ 2743 (22.9%)
Rating 4: ███ 852 (7.1%)
Rating 5: ██ 565 (4.7%)
Rating 6:  0 (0.0%)
Rating 7: █ 295 (2.5%)


In [ ]:
# Group ratings into categories
print("\nInvestment Preference Summary:")
print("-" * 40)

low = sum(1 for r in data if r <= 2)
medium = sum(1 for r in data if 3 <= r <= 5)
high = sum(1 for r in data if r >= 6)

print(f"Low (1-2):    {'😞' * (low // 200)}")
print(f"Medium (3-5): {'😐' * (medium // 200)}")
print(f"High (6-7):   {'😊' * (high // 200)}")
print("\nEach emoji = 500 responses")


Investment Preference Summary:
----------------------------------------
Low (1-2):    😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞😞
Medium (3-5): 😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐😐
High (6-7):   😊

Each emoji = 500 responses


## Conclusion

Through this analysis of 12,000+ investment preferences, we discovered:

- The average Mutual Fund rating is 2.5346, which is below the midpoint 
  of 4, suggesting investors are somewhat skeptical of Mutual Funds.
  
- The median of 2.0000 confirms that more than half of investors gave 
  ratings of 2 or below, showing a general lack of enthusiasm.
  
- The mode of 2 (appearing 6343 times) indicates this is the most 
  common sentiment - mild interest but not strong preference.

This tells us that investors generally view Mutual Funds as a 
lower-priority investment option compared to other vehicles.


### What I Learned

- How to calculate statistics both with pandas and manually
- The importance of understanding what happens "under the hood"
- How to create visualizations without fancy libraries
- Using dictionaries for frequency counting is powerful
